### ISSUE TITLE
Arrow functions don't trigger `jest/no-test-return-statement`

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [jest-community/eslint-plugin-jest#1480]
`jest/valid-describe-callback` does catch arrow functions for `describe` callback, but `jest/no-test-return-statement` doesn't for parametrized tests:

This is caught:
```ts
describe('given decimal places', () => {
  it.each([
    // Minimal example
    { input: 0, decimalPlaces: 0, expected: 0 },
  ])('should round $input with $decimalPlaces decimals to $expected', ({ input, decimalPlaces, expected }) => {
    return expect(roundTo(input, decimalPlaces)).toBe(expected)
  })
})
```

This is not:
```ts
describe('given decimal places', () => {
  it.each([
    // Minimal example
    { input: 0, decimalPlaces: 0, expected: 0 },
  ])('should round $input with $decimalPlaces decimals to $expected', ({ input, decimalPlaces, expected }) =>
    expect(roundTo(input, decimalPlaces)).toBe(expected)
  )
})
```

**Comment by G-Rath** [comment]
Currently this is explicitly allowed: https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/__tests__/no-test-return-statement.test.ts#L15 (and it's in the documentation examples as valid)

I'm not that fussed though arguably changing it now would be a very minor breaking change.

@SimenB do you have any thoughts here?

**Comment by Samuel-Therrien-Beslogic** [comment]
> Currently this is explicitly allowed: https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/__tests__/no-test-return-statement.test.ts#L15 (and it's in the documentation examples as valid)
> 
> I'm not that fussed though arguably changing it now would be a very minor breaking change.
> 
> @SimenB do you have any thoughts here?

Ah, I missed it from the docs.

I'm not the most well-versed in jest code-standards (I mostly turn on the jest rules and follow what they say) so I don't know about all the intricacies behind why certain rules exists (stylistic, consistency, code smells, potential bad understanding of a test, etc.)

I simply noticed something that seemed inconsistent and reported it, so I'll defer to your expertise :)

**Comment by G-Rath** [comment]
I'm going to close this as while personally I'm not a fan of the style I know that at least one person will be using it and I don't think it's worth prematurely supporting an option given Jest itself will error if you actually return anything that isn't a Promise or undefined
